nucypher/network/protocols.py

```diff
@@ -96,4 +116,128 @@ def __radd__(self, other):
         return bytes(other) + bytes(self)
 
     def __repr__(self):
-        return self.uri
+	    return self.uri
+
+
+class HendrixResourceWithStatics(HendrixResource):
```

Comment:
Much of this code feels out of scope to the nucypher codebase and more aligned with hendrix itself.

Reply:
Yeah.  If you want I can add it to Hendrix and we can take it out of here.  I had comments that alluded to doing that someday.  Or we can merge it and use it and then do that as a separate issue.